In [1]:
import pandas as pd
titanic = pd.read_csv("../data/titanic_propre.csv")
titanic

,survived,pclass,sex,age,sibsp,parch,fare,embark_town,alone
0,0,3,male,22.0,1,0,7.2500,Southampton,False
1,1,1,female,38.0,1,0,71.2833,Cherbourg,False
2,1,3,female,26.0,0,0,7.9250,Southampton,True
3,1,1,female,35.0,1,0,53.1000,Southampton,False
4,0,3,male,35.0,0,0,8.0500,Southampton,True
...,...,...,...,...,...,...,...,...,...
886,0,2,male,27.0,0,0,13.0000,Southampton,True
887,1,1,female,19.0,0,0,30.0000,Southampton,True
888,0,3,female,21.5,1,2,23.4500,Southampton,False
889,1,1,male,26.0,0,0,30.0000,Cherbourg,True


In [2]:
survie_par_classe_et_sexe = (titanic.groupby(["pclass", "sex"])["survived"].mean() * 100).round(1)
survie_par_classe_et_sexe

pclass  sex   
1       female    96.8
        male      36.9
2       female    92.1
        male      15.7
3       female    50.0
        male      13.5
Name: survived, dtype: float64

**Réponse 1 :** Oui, dans les trois classes. En 1re classe, 96.8 % des femmes survivent contre 36.9 % des hommes ; en 2e classe 92.1 % contre 15.7 % ; en 3e classe 50.0 % contre 13.5 %. L'écart est énorme partout.

In [3]:
titanic["tranche_age"] = pd.cut(titanic["age"], bins=[0, 12, 17, 59, 100], labels=["Enfant 0-12", "Ado 13-17", "Adulte 18-59", "Senior 60+"])

survie_par_tranche = (titanic.groupby("tranche_age", observed=True)["survived"].mean() * 100).round(1)
survie_par_tranche

tranche_age
Enfant 0-12     58.0
Ado 13-17       47.7
Adulte 18-59    36.4
Senior 60+      26.9
Name: survived, dtype: float64

**Réponse 2 :** Oui. Les enfants de 0 à 12 ans survivent à 58.0 %, les ados à 47.7 %, les adultes à 36.4 % et les plus de 60 ans à 26.9 %. Plus on est jeune, plus la survie est élevée, ce qui suggère que les enfants ont été prioritaires.

In [4]:
prix_moyen_par_port = titanic.groupby("embark_town", observed=True)["fare"].mean().round(2)
prix_moyen_par_port

embark_town
Cherbourg      59.95
Queenstown     13.28
Southampton    27.24
Name: fare, dtype: float64

In [5]:
classes_par_port = (pd.crosstab(titanic["embark_town"], titanic["pclass"], normalize="index") * 100).round(1)
classes_par_port

pclass,1,2,3
embark_town,,,
Cherbourg,50.6,10.1,39.3
Queenstown,2.6,3.9,93.5
Southampton,20.0,25.4,54.6


**Réponse 3 :** Cherbourg a le prix moyen le plus élevé (59.95), devant Southampton (27.24) et Queenstown (13.28). La répartition des classes l'explique en grande partie : 50.6 % des passagers de Cherbourg voyagent en 1re classe, contre 20.0 % à Southampton et seulement 2.6 % à Queenstown, où 93.5 % sont en 3e classe.

In [6]:
survie_selon_solitude = (titanic.groupby("alone")["survived"].mean() * 100).round(1)
survie_selon_solitude

alone
False    50.6
True     30.4
Name: survived, dtype: float64

**Réponse 4 :** Voyager seul diminuait les chances de survie : 30.4 % pour les passagers seuls contre 50.6 % pour ceux accompagnés. Attention, cela ne prouve pas que la solitude soit la cause : les passagers seuls étaient surtout des hommes adultes, un groupe qui a moins survécu.